masakari-controller/controller/masakari_starter.py

```diff
@@ -215,7 +210,8 @@ def add_failed_instance(self, notification_id,
         """
 
         try:
-            db_engine = dbapi.get_engine()
+            self.rc_config.set_record_identifier(notification_id)
```

Comment:
This looks not a thread safe.  The method sets notification_id to the instance referred by rc_config. But all threads share the instance, so  last spin-up thread overwrites _request_store in masakari_config.py if threads >2 works concurrently.

I guess you introduced the identifier for outputting a thread name in log because of my comment[1]. In that case you can specify a thread name[2] where you generate a new thread[3].  threading.current_thread.getName() returns non default thread name, like Thread-1 or Thread-2, only if you specify a thread name.

1. https://github.com/ntt-sic/masakari/pull/43#discussion_r69078837
2. https://docs.python.org/3/library/threading.html#threading.Thread
3. https://github.com/ntt-sic/masakari/blob/master/masakari-controller/controller/masakari_controller.py#L150